pkg/logs/message/message_test.go

```diff
@@ -21,5 +23,98 @@ func TestMessage(t *testing.T) {
 	message.SetContent([]byte("world"))
 	assert.Equal(t, "world", string(message.GetContent()))
 	assert.Equal(t, StatusInfo, message.GetStatus())
+}
+
+func TestNewPayload(t *testing.T) {
+	messages := []*Message{
+		NewMessage([]byte("hello"), nil, "", 0),
+		NewMessage([]byte("world"), nil, "", 0),
+		NewMessage([]byte("test"), nil, "", 0),
+	}
+	encoded := []byte("encoded content")
+	encoding := "gzip"
+	unencodedSize := 100
+
+	payload := NewPayload(messages, encoded, encoding, unencodedSize)
+
+	// Test basic payload properties
+	assert.Equal(t, 3, len(payload.MessageMetas))
+	assert.Equal(t, encoded, payload.Encoded)
+	assert.Equal(t, encoding, payload.Encoding)
+	assert.Equal(t, unencodedSize, payload.UnencodedSize)
+
+	// Test Count method
+	assert.Equal(t, int64(3), payload.Count())
+
+	// Test Size method (each message is 5, 5, and 4 bytes respectively)
+	assert.Equal(t, int64(14), payload.Size())
+}
+func TestPayloadPreservesMessageOrder(t *testing.T) {
+	messages := []*Message{
+		NewMessage([]byte("1"), nil, "", 1),    // datalen = 1
+		NewMessage([]byte("22"), nil, "", 2),   // datalen = 2
+		NewMessage([]byte("333"), nil, "", 3),  // datalen = 3
+		NewMessage([]byte("4444"), nil, "", 4), // datalen = 4
+	}
+	payload := NewPayload(messages, []byte(""), "", 0)
+
+	expectedLengths := []int{1, 2, 3, 4}
+	assert.Equal(t, len(expectedLengths), len(payload.MessageMetas), "Should have same number of message metas")
+
+	for i, msg := range messages {
+		assert.Equal(t, msg.RawDataLen, payload.MessageMetas[i].RawDataLen, "Message at index %d should have RawDataLen of %d", i, msg.GetContent())
+		assert.Equal(t, msg.IngestionTimestamp, payload.MessageMetas[i].IngestionTimestamp, "Message at index %d should have ingestion timestamp %d", i, msg.IngestionTimestamp)
+	}
+}
+
+func TestPayloadAllowsMessageContentGC(t *testing.T) {
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// Create a function to track when our test content gets GC'd
+	wasGCd := false
+	trackGC := func(_ *[]byte) {
+		wasGCd = true
+		wg.Done()
+	}
+
+	var payload *Payload
+
+	// Create scope to allow message to be GC'd
+	func() {
+		// Create message with content we want to track
+		content := make([]byte, 1000000) // Large enough to be noticeable for GC
+		message := NewMessage(content, nil, "", 2)
+
+		// Set up finalizer to track when content is GC'd
+		runtime.SetFinalizer(&message.content, trackGC)
```

Comment:
This is a cool test. 
This seems like a nice way to assert memory behaviors in unit tests and possibly prevent memory regressions in the future 👍